engine/dlib/src/cs/dmsdk/dlib/configfile_gen.cs

```diff
@@ -15,36 +15,103 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
+
+/*# SDK ConfigFile API documentation
+ *
+ * Configuration file access functions.
+ * The configuration file is compiled version of the [file:game.project] file.
+ *
+ * @document
+ * @language C#
+ * @name ConfigFile
+ * @namespace Dlib
+ * @path engine/dlib/src/cs/dmsdk/dlib/configfile_gen.cs
+ */
+
 using System.Runtime.InteropServices;
 using System.Reflection.Emit;
 
 
 namespace dmSDK.Dlib {
     public unsafe partial class ConfigFile
     {
-        /*#
-        * Generated from [ref:ConfigFile]
-        */
+        /// <summary>
+        /// Generated from [ref:ConfigFile]
+        /// </summary>
         [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi, Pack=1)]
         public struct Config
         {
         }
 
-        /*#
-        * Generated from [ref:ConfigFileGetString]
-        */
+        /// <summary>
+        /// Get config value as string, returns default if the key isn't found
+        /// </summary>
+        /// <param name="config" type="HConfigFile">Config file handle</param>
+        /// <param name="key" type="const char*">Key in format section.key (.key for no section)</param>
+        /// <param name="default_value" type="const char*">Default value to return if key isn't found</param>
+        /// <returns name="value" type="const char*">found value or default value</returns>
```

Comment:
However, the "type" attributes are "made up" by me, for keeping data around if anyone wishes to do some plugins later.